physics/cires_ugwp_triggers_v1.F90

```diff
@@ -0,0 +1,576 @@
+            dmax =  dtot
+          endif  
+        enddo
+!	  
+        if (dmax >= tlim_okw) then
+          nf_src = nf_src + 1
+          if_src(i) = 1          
+          taub(i) = tau_min*float(kex)          !* precip(i)/precip_max*coslat(i)
+        endif 
+
+      enddo 
+      print *, '   get_spectra_tau_okwgw '           
+      end subroutine get_spectra_tau_okw   
+!
+!
+!      
+!>\ingroup cires_ugwp_run
+!> @{
+!!
+!!
+      subroutine slat_geos5_tamp_v1(im, tau_amp, xlatdeg, tau_gw)
+!=================
+! GEOS-5 & MERRA-2 lat-dependent GW-source function  tau(z=Zlaunch) =rho*<u'w'>
+!=================
+      implicit none
+      integer :: im     
+      real    :: tau_amp, xlatdeg(im), tau_gw(im)
+      real    :: latdeg, flat_gw, tem
+      integer :: i
+      
+!
+! if-lat
+!
+      do i=1, im
+        latdeg = abs(xlatdeg(i))    
+        if (latdeg < 15.3) then
+          tem = (latdeg-3.0) / 8.0
+          flat_gw = 0.75 * exp(-tem * tem)
+          if (flat_gw < 1.2 .and. latdeg <= 3.0) flat_gw = 0.75
+        elseif (latdeg <  31.0 .and. latdeg >=  15.3) then
+           flat_gw =  0.10
+        elseif (latdeg <  60.0 .and. latdeg >=  31.0) then
+          tem = (latdeg-60.0) / 23.0
+          flat_gw =  0.50 * exp(- tem * tem)
+        elseif (latdeg >=  60.0) then
+          tem = (latdeg-60.0) / 70.0
+          flat_gw =  0.50 * exp(- tem * tem)
+        endif
+        tau_gw(i) = tau_amp*flat_gw 
+      enddo
+!      
+      end subroutine slat_geos5_tamp_v1
+      
+      subroutine slat_geos5(im, xlatdeg, tau_gw)
+!=================
+! GEOS-5 & MERRA-2 lat-dependent GW-source function  tau(z=Zlaunch) =rho*<u'w'>
+!=================
+      implicit none
+      integer :: im     
+      real  :: xlatdeg(im)          
+      real  :: tau_gw(im)
+      real  :: latdeg
+      real, parameter  :: tau_amp = 100.e-3
+      real             :: trop_gw,  flat_gw 
+      integer :: i
+!
+! if-lat
+!
+      trop_gw = 0.75
+      do i=1, im
+      latdeg = xlatdeg(i)    
+      if (-15.3 < latdeg .and. latdeg < 15.3) then
+          flat_gw = trop_gw*exp(-( (abs(latdeg)-3.)/8.0)**2)
+          if (flat_gw < 1.2 .and. abs(latdeg) <= 3.) flat_gw = trop_gw
+      else if (latdeg > -31. .and. latdeg <= -15.3) then
+          flat_gw =  0.10
+      else if (latdeg <  31. .and. latdeg >=  15.3) then
+           flat_gw =  0.10
+      else if (latdeg > -60. .and. latdeg <= -31.) then
+        flat_gw =  0.50*exp(-((abs(latdeg)-60.)/23.)**2)
+      else if (latdeg <  60. .and. latdeg >=  31.) then
+        flat_gw =  0.50*exp(-((abs(latdeg)-60.)/23.)**2)
+      else if (latdeg <= -60.) then
+         flat_gw =  0.50*exp(-((abs(latdeg)-60.)/70.)**2)
+      else if (latdeg >=  60.) then
+         flat_gw =  0.50*exp(-((abs(latdeg)-60.)/70.)**2)
+      end if
+      tau_gw(i) = tau_amp*flat_gw 
+      enddo
+!      
+      end subroutine slat_geos5
+      subroutine init_nazdir(naz,  xaz,  yaz)
+      use ugwp_common_v1 , only : pi2
```

Comment:
Constants should come from the host model through the argument list, if possible.

Reply:
Done.